panoptes_client/caesar.py

```diff
@@ -68,20 +78,36 @@ def get_extracts_by_workflow_and_subject(self, workflow_id, subject_id):
         return self.http_get(
             f'workflows/{workflow_id}/extractors/extractor/extracts', params={'subject_id': subject_id})[0]
 
-    def add_workflow(self, workflow_id, public_extracts=False, public_reductions=False):
+    def save_workflow(self, workflow_id, public_extracts=False, public_reductions=False):
         """
-        Adds workflow with provided workflow_id to Caesar. Returns workflow as a dict from Caesar if successful.
+        Adds/updates workflow with provided workflow_id to Caesar. Checks to see if workflow exists in Caesar, if not
+        then creates workflow and returns workflow as a dict from Caesar if created.
+        If workflow is already in Caesar, will update the Caesar workflow.
 
         Examples::
-            Caesar().add_workflow(123, public_extracts=True, public_reductions=True)
-        """
-        return self.http_post('workflows', json={
-            'workflow': {
-                'id': workflow_id,
-                'public_extracts': public_extracts,
-                'public_reductions': public_reductions
-            }
-        })[0]
+            Caesar().save_workflow(123, public_extracts=True, public_reductions=True)
+        """
+        try:
+            self.get_workflow(workflow_id)
+        except PanoptesAPIException as err:
+            if "couldn't find workflow with 'id'" in str(err).lower():
+                return self.http_post('workflows', json={
+                    'workflow': {
+                        'id': workflow_id,
+                        'public_extracts': public_extracts,
+                        'public_reductions': public_reductions
+                    }
+                })[0]
+            else:
+                raise err
+        else:
+            return self.http_put(f'workflows/{workflow_id}', json={
```

Comment:
Are there any unintended consequences resulting from the choice here to _always_ submit a PUT request, even if there is no update (i.e., workflow already exists and public_extracts / public_reductions remain unchanged from current state)?  I'm wondering if it is worth adding a check regarding agreement between current and desired state of public extracts and reductions or whether keeping the simpler case of always PUTing the desired state (as written) is preferred.